app/src/main/java/org/thoughtcrime/securesms/onboarding/LoadingActivity.kt

```diff
@@ -0,0 +1,121 @@
+package org.thoughtcrime.securesms.onboarding
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.activity.viewModels
+import androidx.compose.animation.core.Animatable
+import androidx.compose.animation.core.TweenSpec
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.res.stringResource
+import androidx.lifecycle.lifecycleScope
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.launch
+import network.loki.messenger.R
+import org.session.libsession.utilities.AppTextSecurePreferences
+import org.session.libsession.utilities.TextSecurePreferences
+import org.thoughtcrime.securesms.ApplicationContext
+import org.thoughtcrime.securesms.BaseActionBarActivity
+import org.thoughtcrime.securesms.dependencies.ConfigFactory
+import org.thoughtcrime.securesms.onboarding.messagenotifications.startPNModeActivity
+import org.thoughtcrime.securesms.onboarding.pickname.startPickDisplayNameActivity
+import org.thoughtcrime.securesms.ui.AppTheme
+import org.thoughtcrime.securesms.ui.ProgressArc
+import org.thoughtcrime.securesms.ui.contentDescription
+import org.thoughtcrime.securesms.util.setUpActionBarSessionLogo
+import javax.inject.Inject
+
+private const val EXTRA_MNEMONIC = "mnemonic"
+
+@AndroidEntryPoint
+class LoadingActivity: BaseActionBarActivity() {
+
+    @Inject
+    lateinit var configFactory: ConfigFactory
+
+    @Inject
+    lateinit var prefs: TextSecurePreferences
+
+    private val viewModel: LoadingViewModel by viewModels()
+
+    @Deprecated("Deprecated in Java")
+    override fun onBackPressed() {
+        return
+    }
+
+    private fun register(skipped: Boolean) {
+        prefs.setLastConfigurationSyncTime(System.currentTimeMillis())
+
+        val flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
+
+        when {
+            skipped -> startPickDisplayNameActivity(true, flags)
+            else -> startPNModeActivity(flags)
```

Comment:
What does PN stand for? Could we use a more descriptive name?

Reply:
fixed